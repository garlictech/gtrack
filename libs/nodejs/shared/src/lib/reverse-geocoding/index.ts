export * from './reverse-geocoding.service';
