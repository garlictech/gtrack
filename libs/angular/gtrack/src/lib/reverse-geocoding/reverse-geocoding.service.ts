import {
  googleReverseGeocodingService,
  Deps,
} from '@bit/garlictech.universal.gtrack.google-maps';
import * as OE from 'fp-ts-rxjs/lib/ObservableEither';
import { Injectable, Inject } from '@angular/core';
import {
  GOOGLE_API_CONFIG,
  GoogleApiConfig,
} from '@bit/garlictech.angular.gtrack.google-maps';

@Injectable({
  providedIn: 'root',
})
export class ReverseGeocodingService {
  private deps: Deps;

  constructor(@Inject(GOOGLE_API_CONFIG) config: GoogleApiConfig) {
    this.deps = {
      apiKey: config.apiKey,
      client: undefined,
    };
  }

  getCityFromGoogle<POINT extends { lat: number; lon: number }>(
    point: POINT
  ): OE.ObservableEither<any, string> {
    return googleReverseGeocodingService(this.deps).getCity(point);
  }
}
