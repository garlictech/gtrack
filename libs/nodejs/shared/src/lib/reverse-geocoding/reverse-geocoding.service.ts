import { googleReverseGeocodingService } from '@bit/garlictech.universal.gtrack.google-maps';
import * as OE from 'fp-ts-rxjs/lib/ObservableEither';
import { googleMapsDeps } from '@bit/garlictech.nodejs.shared.google-maps';

export const getCityFromGoogle = <POINT extends { lat: number; lon: number }>(
  point: POINT
): OE.ObservableEither<any, string> =>
  googleReverseGeocodingService(googleMapsDeps).getCity(point);
