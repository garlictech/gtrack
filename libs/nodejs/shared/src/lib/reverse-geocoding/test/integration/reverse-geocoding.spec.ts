import { tap } from 'rxjs/operators';
import { getCityFromGoogle } from '../../reverse-geocoding.service';

describe('Reverse geocoding service tests', () => {
  it('Get addres of a point with the google map service', done => {
    getCityFromGoogle({ lat: 48, lon: 19 })
      .pipe(tap(res => expect(res).toMatchSnapshot()))
      .subscribe(() => done());
  });
});
