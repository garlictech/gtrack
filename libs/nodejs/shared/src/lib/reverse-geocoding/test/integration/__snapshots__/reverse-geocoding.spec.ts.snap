// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reverse geocoding service tests Get addres of a point with the google map service 1`] = `
Object {
  "_tag": "Right",
  "right": "Bernecebar√°ti, 2639",
}
`;
